app/models/reviewer_access.rb

```diff
@@ -0,0 +1,94 @@
+# manage anonymous access to private studies via access code & pin
+# will generate a short-lived session for reviewers with valid credentials
+class ReviewerAccess
+  include Mongoid::Document
+  include Mongoid::Timestamps
+  include ActiveSupport::SecurityUtils
+
+  field :access_code, type: String # UUID
+  field :pin, type: String # 6 character string
+  field :expires_at, type: Date, default: 2.months.from_now.to_date
+
+  PIN_LENGTH = 10
+  # regex to strip all non-alphanumeric characters from a user-supplied pin value
+  PIN_SANITIZER = /(\W|_)/.freeze
+  # regex to validate a session_key is a UUID
+  UUID_REGEX = /^[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$/.freeze
```

Comment:
seems like there should be a ruby helper method for this

Reply:
Found the `uuid` gem to do this (it looked like it wasn't maintained, but if we only use it for `UUID.validate()` it should be fine).

Reply:
446268b